web/components/application/hacker/pages/PageTemplate.js

```diff
@@ -46,15 +70,18 @@ export class HackerApplicationPageTemplate extends React.Component {
               onClick={activeIndex === 0 ? cancelApplication : onPageBack}
             />
             <PrimaryButton
-              disabled={!isNextButtonEnabled}
-              text={getPrimaryButtonText(activeIndex, count)}
-              onClick={activeIndex !== count - 1 ? onPageNext : () => this.captcha.execute()}
+              disabled={!isNextButtonEnabled || isSubmitting}
+              text={isSubmitting ? 'Submitting...' : getPrimaryButtonText(activeIndex, count)}
+              onClick={activeIndex !== count - 1 ? onPageNext : () => this.onSubmitButtonClick()}
             />
+            {submitFailed ? (<p>Failed to submit application, please try again!</p>) : null}
             <Reaptcha
               ref={e => (this.captcha = e)}
               sitekey={recaptchaConfig[process.env.NODE_ENV].sitekey}
               onVerify={(token) => {
-                submitApplication(null, token);
+                this.setState({ isSubmitting: true });
+                this.setState({ token });
```

Comment:
Why is this necessary?

Reply:
If we're in this function it means that the token was successfully generated but the call to our cloud function may still fail. In this case, the recaptcha refuses to generate another token so I use the same token in L22 when the user tries to resubmit.